sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q23b/simplified.txt

```diff
@@ -188,22 +188,6 @@ TakeOrderedAndProject [c_last_name,c_first_name,sales]
                                       InputAdapter
                                         ReusedExchange [c_customer_sk,sum,isEmpty] #8
                         InputAdapter
-                          BroadcastExchange #19
-                            WholeStageCodegen (33)
-                              SortMergeJoin [c_customer_sk,c_customer_sk]
-                                InputAdapter
-                                  WholeStageCodegen (29)
-                                    Sort [c_customer_sk]
-                                      InputAdapter
-                                        ReusedExchange [c_customer_sk,c_first_name,c_last_name] #15
-                                InputAdapter
-                                  WholeStageCodegen (32)
-                                    Sort [c_customer_sk]
-                                      Project [c_customer_sk]
-                                        Filter [sum(CheckOverflow((promote_precision(cast(cast(ss_quantity as decimal(10,0)) as decimal(12,2))) * promote_precision(cast(ss_sales_price as decimal(12,2)))), DecimalType(18,2), true))]
-                                          ReusedSubquery [tpcds_cmax] #3
-                                          HashAggregate [c_customer_sk,sum,isEmpty] [sum(CheckOverflow((promote_precision(cast(cast(ss_quantity as decimal(10,0)) as decimal(12,2))) * promote_precision(cast(ss_sales_price as decimal(12,2)))), DecimalType(18,2), true)),sum(CheckOverflow((promote_precision(cast(cast(ss_quantity as decimal(10,0)) as decimal(12,2))) * promote_precision(cast(ss_sales_price as decimal(12,2)))), DecimalType(18,2), true)),sum,isEmpty]
-                                            InputAdapter
-                                              ReusedExchange [c_customer_sk,sum,isEmpty] #8
+                          ReusedExchange [c_customer_sk,c_first_name,c_last_name] #19
```

Comment:
After the fix, more exchange reuse happens :)

Seems the problem was caused by `ReusedSubquery`.